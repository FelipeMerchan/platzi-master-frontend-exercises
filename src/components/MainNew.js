import React from 'react'

import './styles/MainNew.css'
import noticeImage from '../images/barack-obama.png'

function MainNew() {
  return (
    <section className="MainNew">
      <figure>
        <img className="w-100" width="600" src={ noticeImage } alt="Barack Obama" />
        <figcaption>Lorem ipsum dolor sit amet, consectetuer</figcaption>
        <p className="text-justify">Lorem ipsum dolor sit amet, 
          consectetuer adipiscing elit, sed diam nonummy nibh 
          euismod tincidunt ut laoreet dolore magna aliquam erat 
          volutpat. Ut wisi enim ad minim veniam
        </p>
      </figure>
    </section>
  )
}

export default MainNew