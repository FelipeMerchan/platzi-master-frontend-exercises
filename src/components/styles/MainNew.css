.MainNew {
  padding-bottom: var(--space);
  border-bottom: 1px solid var(--black-color);
}