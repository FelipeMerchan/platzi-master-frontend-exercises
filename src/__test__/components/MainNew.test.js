import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import MainNew from '../../components/MainNew'

describe('Probar el componente <MainNew />', () => {
  const mainNew = shallow(<MainNew />)

  test('Render del componente MainNew', () => {
    expect(mainNew.length).toEqual(1)
  })
  test('Render del caption', () => {
    expect(mainNew.find('figcaption').text()).toEqual('Lorem ipsum dolor sit amet, consectetuer')
  })
  test('Render con el className text-justify', () => {
    expect(mainNew.find('p').hasClass('text-justify')).toBe(true)
  })
})

describe('MainNew Snapshot', () => {
  test('Comprobar la UI del componente MainNew', () => {
    const mainNew = create(<MainNew />)
    expect(mainNew.toJSON()).toMatchSnapshot()
  })
})