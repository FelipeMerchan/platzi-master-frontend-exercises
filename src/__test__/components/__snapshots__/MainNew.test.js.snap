// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNew Snapshot Comprobar la UI del componente MainNew 1`] = `
<section
  className="MainNew"
>
  <figure>
    <img
      alt="Barack Obama"
      className="w-100"
      src="barack-obama.png"
      width="600"
    />
    <figcaption>
      Lorem ipsum dolor sit amet, consectetuer
    </figcaption>
    <p
      className="text-justify"
    >
      Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam
    </p>
  </figure>
</section>
`;
